Test1/src/leetcode/LeetCode17.java

```diff
@@ -0,0 +1,70 @@
+package leetcode;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * https://leetcode.com/problems/letter-combinations-of-a-phone-number/
+ * @author minibig
+ */
+public class LeetCode17 {
+
+	public List<String> letterCombinations(String digits) {
+		
+		List<String> list = new ArrayList<String>();
+		
+		if(digits.isEmpty()) return list;
+		
+		Map<Integer, String> digitMap = new HashMap<Integer, String>();
+		
+		digitMap.put(2, "abc");
+		digitMap.put(3, "def");
+		digitMap.put(4, "ghi");
+		digitMap.put(5, "jkl");
+		digitMap.put(6, "mno");
+		digitMap.put(7, "pqrs");
+		digitMap.put(8, "tuv");
+		digitMap.put(9, "wxyz");
+		
+		int digitsLength = digits.length();
+		
+		String[] str = new String[digitsLength];
```

Comment:
String[] str = new String[digits.length()];

You don't need the variable `digitsLength`.

`digits.length()` is very trivial and straightforward.